### PR TITLE
Validate malformed endpoint

### PR DESCRIPTION
**Pull request by shanakama**
### Purpose
The PR contains fixes for validate malformed endpoint.Generally validating the hostname of the endpoint.

### Issues
Fixes : https://github.com/wso2/product-microgateway/issues/2183

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments
Tested locally

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by Rajith90** [comment]
Shall we add a unit and integration test for this,

File: adapter/internal/oasparser/model/open_api.go [hunk 1]
```diff
@@ -29,6 +30,10 @@ import (
 	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
 )
 
+const (
+	hostNameValidator = "^[a-zA-Z0-9]+[a-zA-Z0-9-.]+[0-9a-zA-Z]$"
+)
```
Comment (Rajith90) [inline adapter/internal/oasparser/model/open_api.go:35]: Can you add a comment explaining the regex and with few samples, and few wrong samples

Comment (VirajSalaka) [reply adapter/internal/oasparser/model/open_api.go:35]: I think this is the specification 
https://datatracker.ietf.org/doc/html/rfc952

Comment (shanakama) [reply adapter/internal/oasparser/model/open_api.go:52]: Comment can find here : 018832e

**Comment by shanakama** [comment]
Unit tests added

**Comment by shanakama** [comment]
> Shall we add a unit and integration test for this,

Unit tests added. I think , no need Integration tests for this . Unit tests are enough.

**Review by Rajith90 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20210729.8

**Comment by choreo-cicd** [comment]
[succeeded] stage-deployment : 20210730.2

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20210802.1

**Pull request merged**